### PR TITLE
Add binaries and support for macOS

### LINKED ISSUE #4: Add support for MacOS
In `audb` we are planning to support Linux, Windows and MacOS, so we need to add support for MacOS here as well.

### PR DESCRIPTION
**Pull request by chausner-audeering**
Fixes #4 by adding binaries for macOS. I guess we should test it first before merging, I am not sure if the rpath patching on the library worked.

### CODE REVIEW

**Comment by hagenw** [comment]
For testing we can simply enable the macOS test in `.github/workflows/test.yml`

**Comment by chausner-audeering** [comment]
> For testing we can simply enable the macOS test in `.github/workflows/test.yml`

I'm surprised but it seems it already works. :) I would assume the tests would fail if there were missing dependencies like libsndfile on Ubuntu?

We should mention the system requirements somewhere in the documentation. In particular, the set of supported operating systems. Also the fact that 64-bit is required because the binaries won't run on x86. The Windows binary should be very compatible and supported on all versions of Windows 64-bit. The Linux binaries should run on Debian and Ubuntu but finding out which versions are supported is not trivial. At least 18.04 is supported but 16.04 might not. For macOS, I am not sure but I assume the compatibility story is similar to Linux. Maybe @aud-ktw knows more about which versions of macOS can run binaries that have been compiled in our CI?

**Comment by chausner-audeering** [comment]
@agfcrespi I got the macOS binaries from our CI after adjusting the internal repo of audresamplelib. Only afterwards I saw that we have also open-sourced it at https://github.com/audeering/audresamplelib. I assume the two repos should be synchronized or we should lock the internal repository.

**Comment by hagenw** [comment]
> I would assume the tests would fail if there were missing dependencies like libsndfile?

I think `libsndfile` comes already with macOS, at least for tests in `audformat` we have to install it for Ubtunu, but also not for macOS, compare https://github.com/audeering/audformat/blob/master/.github/workflows/test.yml#L48-L50

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -19,6 +19,10 @@ jobs:
             python-version: 3.7
           - os: ubuntu-latest
             python-version: 3.8
+          - os: macos-latest
+            python-version: 3.7
+          - os: macos-latest
+            python-version: 3.8
```
Comment (hagenw) [inline .github/workflows/test.yml:25]: I would stick to our conservative approach and test macOS only for Python 3.6 as we do in `audformat`:

```yaml
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ ubuntu-18.04, windows-latest, macOS-latest ]
        python-version: [3.6]
        include:
          - os: ubuntu-latest
            python-version: 3.7 
          - os: ubuntu-latest
            python-version: 3.8
```

**Review by hagenw — CHANGES_REQUESTED** [review]
As we now have different binaries for different platforms, we might also think about [publishing different wheels for different platforms](https://packaging.python.org/guides/distributing-packages-using-setuptools/#platform-wheels) in order to stay with a small wheel size.

The only downside is, I have no clue how to force `python setup.py sdist bdist_wheel` to build for selected operating systems.

**Comment by agfcrespi** [comment]
> @agfcrespi I got the macOS binaries from our CI after adjusting the internal repo of audresamplelib. Only afterwards I saw that we have also open-sourced it at https://github.com/audeering/audresamplelib. I assume the two repos should be synchronized or we should lock the internal repository.

@hagenw which approach would you suggest? Sync the two repos or archive the internal one? I am personally leaning towards the latter.

**Comment by hagenw** [comment]
> > @agfcrespi I got the macOS binaries from our CI after adjusting the internal repo of audresamplelib. Only afterwards I saw that we have also open-sourced it at https://github.com/audeering/audresamplelib. I assume the two repos should be synchronized or we should lock the internal repository.
> 
> @hagenw which approach would you suggest? Sync the two repos or archive the internal one? I am personally leaning towards the latter.

Yes, please archive the internal one as it will always lead to a mess if we have two different repos for the same thing.

File: .github/workflows/test.yml (hunk 1 above)
Comment (chausner-audeering) [reply .github/workflows/test.yml]: OK sure, I changed it: 96d8ba1e9b0e025ed03c01eadbc946a398c1498b

**Comment by hagenw** [comment]
> > As we now have different binaries for different platforms, we might also think about [publishing different wheels for different platforms](https://packaging.python.org/guides/distributing-packages-using-setuptools/#platform-wheels) in order to stay with a small wheel size.
> 
> I agree that would be nice. However, it would be great if we could look into this separately and get macOS support into audresample quickly because we are a bit blocked by this (we want to run audb in our internal CI and need the builds on all platforms).

OK, I agree. I created https://github.com/audeering/audresample/issues/11 to track the wheels issue and will merge this now.

Do you want to fix the Windows issue as well (https://github.com/audeering/audresample/pull/5), or should we made a new release first that just supports macOS?

**Review by hagenw — APPROVED** [review]

**Pull request merged**